pkg/scheduler/ugm/queue_tracker.go

```diff
@@ -85,6 +87,7 @@ const (
 	group
 )
 
+// Note: Lock free call. The Lock of the linked tracker(UserTracker/GroupTracker) should be held before calling this function.
```

Comment:
Do we need to add comments for all methods? it seems only `headroom` and `canRunApp` need those comments since the method "naming" could misdirect us.

Reply:
Hi @chia7712, 
The comments were added to ensure that any new functions in GroupTracker/UserTracker will acquire lock before calling this lock-required functions in QueueTracker . It's not only for headroom and canRunApp.